script.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask, request, jsonify
+import mysql.connector
+from dotenv import load_dotenv
+import os
+
+app = Flask(__name__)
+load_dotenv()
+
+def connect_to_db():
+    """
+    A function that creates a connection to db.
+    """
+    return mysql.connector.connect(
+        host=os.getenv("MYSQL_HOST"),
+        user=os.getenv("MYSQL_ROOT_USER"),
+        password=os.getenv("MYSQL_ROOT_PASSWORD"),
+        database=os.getenv("MYSQL_DATABASE"),
+        port=3306
+    )
+
+@app.route('/<table_name>', methods=['POST'])
+def add_data(table_name):
+    """
+    A function that adds a new user to db.
+    """
+    try:
+        mysql_db = connect_to_db() #creates a connection to db
+        mycursor = mysql_db.cursor() 
+        input_name = request.json.get("name")
+        checks_table = f"SHOW TABLES LIKE '{table_name}';" # checks if the given table is in db
+        mycursor.execute(checks_table)
+        result = mycursor.fetchone()
+        if not result: #if the table is not in the db, it creates a new table
```

Comment:
if you have only one table, you can add in once, no need to try to create each request